grease-exe/src/Grease/Main.hs

```diff
@@ -1209,17 +1251,21 @@ loadAddrOvs ::
   ( Symbolic.SymArchConstraints arch
   , ?parserHooks :: CSyn.ParserHooks (Symbolic.MacawExt arch)
   ) =>
+  GreaseLogAction ->
   ArchContext arch ->
   C.HandleAllocator ->
   MM.Memory (MC.ArchAddrWidth arch) ->
   SimOpts ->
   IO (AddressOverrides arch)
-loadAddrOvs archCtx halloc memory simOpts = do
+loadAddrOvs la archCtx halloc memory simOpts = do
   mbAddrOvs <- loadAddressOverrides (regStructRepr archCtx) halloc memory (simAddressOverrides simOpts)
   case mbAddrOvs of
-    Left err -> do
-      let msg = PP.renderStrict (PP.layoutPretty PP.defaultLayoutOptions (PP.pretty err))
-      throw (GreaseException ("user error: " <> msg))
+    -- See Note [Explicitly listed errors]
+    Left e@AddrOv.BadAddress{} -> userError la (PP.pretty e)
```

Comment:
not sure if it save much but could maybe bind a local function (Pretty e) =>  e -> IO unit while still matching over all of them that closes over la as well